projects/components/src/table/cells/state-parsers/table-cell-state-checkbox.ts

```diff
@@ -0,0 +1,17 @@
+import { StatefulTableRow, TableRowState } from '../../table-api';
+import { TableCellParser } from '../table-cell-parser';
+import { TableCellParserBase } from '../table-cell-parser-base';
+import { CoreTableCellParserType } from '../types/core-table-cell-parser-type';
+
+@TableCellParser({
+  type: CoreTableCellParserType.Checkbox
+})
+export class TableCellStateParserCheckbox extends TableCellParserBase<TableRowState, TableRowState, undefined> {
+  public parseValue(_: TableRowState, rowData: StatefulTableRow): TableRowState {
+    return rowData.$$state;
```

Comment:
I'm not sure how this is working. Shouldn't you need to map this to the specific property you are looking for? In this case, the boolean for `selected`?

Reply:
This approach looks right, so if its working all good - approved. Just curious where the final mapping to the property is happening.

Reply:
So, the renderer actually depends on the entire state. So i didn't change that part. Just using the row to find state object instead of depending on the cell value and deal with id conflicts

Reply:
Cool. Run with it.